packages/lib/CalendarService.ts

```diff
@@ -190,20 +190,44 @@ export default abstract class BaseCalendarService implements Calendar {
 
       const eventsToUpdate = events.filter((e) => e.uid === uid);
       return Promise.all(
-        eventsToUpdate.map((e) => {
+        eventsToUpdate.map((eventItem) => {
+          event = eventItem;
           return updateCalendarObject({
             calendarObject: {
-              url: e.url,
-              data: iCalString,
-              etag: e?.etag,
+              url: event.url,
+              data: iCalString?.replace(/METHOD:[^\r\n]+\r\n/g, ""),
+              etag: event?.etag,
             },
             headers: this.headers,
           });
         })
-      ).then((p) => p.map((r) => r.json() as unknown as NewCalendarEventType));
+      ).then((responses) =>
+        responses.map((response) => {
+          if (response.status > 199 && response.status < 207) {
```

Comment:
Confirms if the update call went through successfully. Some providers return updated data while the others choose not to (sending 204 instead)